features/paginering.feature

```diff
@@ -0,0 +1,78 @@
+
+  Scenario: de zoekvraag levert meerdere pagina's en met de page parameter wordt de eerste pagina gevraagd
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467&page=1
+    Dan zitten er 20 adressen in het antwoord
+    En zijn dit adressen 1 tot en met 20 met deze pandidentificatie
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=1"
+    En bevat het antwoord geen attribuut _links.first
+    En bevat het antwoord geen attribuut _links.previous
+    En is attribuut _links.next.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=2"
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467"
+
+  Scenario: de zoekvraag levert meerdere pagina's en met de page parameter wordt een volgende pagina gevraagd
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467&page=3
+    Dan zitten er 20 adressen in het antwoord
+    En zijn dit adressen 41 tot en met 60 met deze pandidentificatie
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=3"
+    En is attribuut _links.first.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=1"
+    En is attribuut _links.previous.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=2"
+    En is attribuut _links.next.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=4"
+
+  Scenario: de zoekvraag levert meerdere pagina's en met de page parameter wordt de laatste pagina gevraagd
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467&page=4
+    Dan zitten er 12 adressen in het antwoord
+    En zijn dit adressen 61 tot en met 72 met deze pandidentificatie
+    En is attribuut _links.first.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=1"
+    En is attribuut _links.previous.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=3"
+    En bevat het antwoord geen attribuut _links.next
+
+  Scenario: de zoekvraag levert meerdere pagina's en met de page parameter wordt een pagina bevraagd die niet bestaat
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000467&page=15
+    Dan zitten er 12 adressen in het antwoord
+    En is attribuut _links.self.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=15"
+    En is attribuut _links.first.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=1"
+    En is attribuut _links.previous.href gelijk aan "/adressen?pandidentificatie=0826100000000467&page=3"
+    En bevat het antwoord geen attribuut _links.next
+
+  Scenario: de zoekvraag levert één pagina
+    Als de request wordt gedaan naar /adressen?pandidentificatie=0826100000000471
+    Dan zitten er 2 adressen in het antwoord
+    En bevat het antwoord geen attribuut _links.first
```

Comment:
Als er maar één pagina wordt teruggegeven, dan is dat ook de first pagina, first zou er dan wel moeten zijn. Het is niet dat er dan geen first pagina is, maar dat wordt wel gesuggereerd als first wordt weggelaten.

Reply:
zie eerdere opmerking over first